### PR TITLE
add /api/providers/#/metric_rollups

### PR DESCRIPTION
**Pull request by kbrock**
Fixes https://github.com/ManageIQ/manageiq/issues/22827

### CODE REVIEW

File: spec/requests/providers_spec.rb [hunk 1]
```diff
@@ -1453,6 +1453,44 @@ def gen_import_request
...
+
+      expected = {
+        'count'    => 5,
+        'subcount' => 3,
+        'pages'    => 1
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+      expect(response.parsed_body['links'].keys).to match_array(%w(self first last))
+#?      expect(response.parsed_body).to include("resources" => [])
```
Comment (Fryguy) [inline spec/requests/providers_spec.rb:1482]: Do you want to keep this before merge?  Seems like it was temporary commented.

File: spec/requests/providers_spec.rb [hunk 2]
```diff
@@ -1453,6 +1453,44 @@ def gen_import_request
...
+      get(url, :params => { :capture_interval => 'hourly', :start_date => Time.zone.today.to_s })
+
+      expected = {
+        'count'    => 5,
+        'subcount' => 3,
+        'pages'    => 1
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+      expect(response.parsed_body['links'].keys).to match_array(%w(self first last))
```
Comment (Fryguy) [inline spec/requests/providers_spec.rb:1481]: Super minor rubocop

```suggestion
      expect(response.parsed_body['links'].keys).to match_array(%w[self first last])
```

File: spec/requests/providers_spec.rb [hunk 3]
```diff
@@ -1453,6 +1453,44 @@ def gen_import_request
...
+      FactoryBot.create_list(:metric_rollup_cm_hr, 3, :resource => provider, :timestamp => 1.hour.ago)
+      FactoryBot.create_list(:metric_rollup_cm_daily, 1, :resource => provider)
+      FactoryBot.create_list(:metric_rollup_cm_hr, 1, :resource => provider1)
+    end
+
+    it 'returns the metric rollups for the vm' do
+      api_basic_authorize subcollection_action_identifier(:providers, :metric_rollups, :read, :get)
+
+      provider.metric_rollups.where(:capture_interval_name => 'hourly').name
+      byebug
```
Comment (Fryguy) [inline spec/requests/providers_spec.rb:1471]: oops

```suggestion
      byebug
```

**Comment by Fryguy** [comment]
Overall LGTM - seems some debug code snuck into the commit

File: config/api.yml [hunk 4]
```diff
@@ -2904,6 +2905,12 @@
         :identifier:
         - ems_infra_show
         - ems_storage_show
+    :metric_rollups_subcollection_actions:
+      :get:
+      - :name: read
+        :identicier:
```
[the remark this answers is not in the document]
Comment (kbrock) [reply config/api.yml:2911]: Then this makes me worry about the test implementations

**Comment by kbrock** [comment]
update:
- fixed cop `%[]`
- fixed role name typo
- fixed `identifiers` typo
- removed byebug
- removed commented out test line

File: spec/requests/providers_spec.rb [hunk 5]
```diff
@@ -1453,6 +1453,42 @@ def gen_import_request
...
+    let(:provider1) { FactoryBot.create(:ems_redhat) }
+    let(:url) { api_provider_metric_rollups_url(nil, provider) }
+
+    before do
+      FactoryBot.create_list(:metric_rollup_cm_hr, 3, :resource => provider, :timestamp => 1.hour.ago)
+      FactoryBot.create_list(:metric_rollup_cm_daily, 1, :resource => provider)
+      FactoryBot.create_list(:metric_rollup_cm_hr, 1, :resource => provider1)
+    end
+
+    it 'returns the metric rollups for the vm' do
```
Comment (Fryguy) [inline spec/requests/providers_spec.rb:1467]:
```suggestion
    it 'returns the metric rollups for the provider' do
```

**Review by Fryguy — APPROVED** [review]
One minor typo and this is good to merge.

**Comment by kbrock** [comment]
update:
- `s/vm/provider` -- copy paste sure is great. until it is not

**Pull request merged**